packages/plugin-core/src/views/GraphPanel.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  DendronTreeViewKey,
+  DMessage,
+  DMessageEnum,
+  GraphEvents,
+  GraphViewMessage,
+  GraphViewMessageEnum,
+  NoteProps,
+  NoteUtils,
+  OnDidChangeActiveTextEditorMsg,
+  VSCodeEvents,
+} from "@dendronhq/common-all";
+import { MetadataService, WorkspaceUtils } from "@dendronhq/engine-server";
+import _ from "lodash";
+import * as vscode from "vscode";
+import { GotoNoteCommand } from "../commands/GotoNote";
+import { IDendronExtension } from "../dendronExtensionInterface";
+import { Logger } from "../logger";
+import { GraphStyleService } from "../styles";
+import { AnalyticsUtils } from "../utils/analytics";
+import { VSCodeUtils } from "../vsCodeUtils";
+import { WebViewUtils } from "./utils";
+
+export class GraphPanel implements vscode.WebviewViewProvider {
+  public static readonly viewType = DendronTreeViewKey.GRAPH_PANEL;
+  private _view?: vscode.WebviewView;
+  private _ext: IDendronExtension;
+  constructor(extension: IDendronExtension) {
+    this._ext = extension;
+    this._ext.context.subscriptions.push(
+      vscode.window.onDidChangeActiveTextEditor(this.onOpenTextDocument, this)
+    );
+  }
+
+  public postMessage(msg: DMessage) {
+    this._view?.webview.postMessage(msg);
+  }
+
+  public async resolveWebviewView(
+    webviewView: vscode.WebviewView,
+    _context: vscode.WebviewViewResolveContext<unknown>,
+    _token: vscode.CancellationToken
+  ) {
+    this._view = webviewView;
+    await WebViewUtils.prepareTreeView({
+      ext: this._ext,
+      key: DendronTreeViewKey.GRAPH_PANEL,
+      webviewView,
+    });
+    webviewView.webview.onDidReceiveMessage(
+      this.onDidReceiveMessageHandler,
+      this
+    );
+
+    webviewView.onDidChangeVisibility(() => {
+      AnalyticsUtils.track(VSCodeEvents.GraphPanelClicked, {
```

Comment:
To reduce the number of types of events, can we use the `GraphPanelUsed` event here as well, set the type property to `VisibilityChanged` and have a state property?

Similar to what I did for Backlinks Panel: 

```typescript
    backlinkTreeView.onDidChangeVisibility((e) => {
      AnalyticsUtils.track(VSCodeEvents.BacklinksPanelUsed, {
        type: "VisibilityChanged",
        state: e.visible ? "Visible" : "Collapsed",
      });
    });
```